### ISSUE TITLE
Using feed_dict is more than 5x faster than using dataset API?

### ISSUE THREAD

**Issue by codescv** [tensorflow/tensorflow#19933]
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS 10.13.4 (on Linux Ubuntu 16.04 you get similar results)
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7 cpu
- **Python version**:  3.6.5
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
I created a dataset in TFRecord format for testing. Every entry contains 200 columns, named C1 - C199, each being a strings list, and a label column to denote the labels. The code to create the data can be found here: https://github.com/codescv/tf-dist/blob/8bb3c44f55939fc66b3727a730c57887113e899c/src/gen_data.py#L25

Then I used a linear model to train the data. The first approach looks like this:

```
dataset = tf.data.TFRecordDataset(data_file)
dataset = dataset.prefetch(buffer_size=batch_size*10)
dataset = dataset.map(parse_tfrecord, num_parallel_calls=5)
dataset = dataset.repeat(num_epochs)
dataset = dataset.batch(batch_size)

features, labels = dataset.make_one_shot_iterator().get_next()    
logits = tf.feature_column.linear_model(features=features, feature_columns=columns, cols_to_vars=cols_to_vars)
train_op = ...

with tf.Session() as sess:
    sess.run(train_op)
```
The full code can be found here: https://github.com/codescv/tf-dist/blob/master/src/lr_single.py

When I run the code above, I get 0.85 steps/sec (batch size being 1024).

In the second approach, I manually get batches from Dataset into python, then feed them to a placeholder, like this:

```
example = tf.placeholder(dtype=tf.string, shape=[None])
features = tf.parse_example(example, features=tf.feature_column.make_parse_example_spec(columns+[tf.feature_column.numeric_column('label', dtype=tf.float32, default_value=0)]))
labels = features.pop('label')
train_op = ...

dataset = tf.data.TFRecordDataset(data_file).repeat().batch(batch_size)
next_batch = dataset.make_one_shot_iterator().get_next()

with tf.Session() as sess:
    data_batch = sess.run(next_batch)
    sess.run(train_op, feed_dict={example: data_batch})
```
The full code can be found here: https://github.com/codescv/tf-dist/blob/master/src/lr_single_feed.py

When I run the code above, I get 5 steps/sec. That is 5x faster than the first approach. This is what I do not understand, because theoretically the second should be slower due to the extra serialization/deserialization of data batches.

Is this possibly a bug or am I using it mistakenly ?

Thanks!

### Source code / logs
I have also included some profile traces below:

(using tf Dataset API)
![lr_single](https://user-images.githubusercontent.com/124190/41281915-3b049342-6e65-11e8-8b04-4d4611a96fa8.png)

[profile-101.json.zip](https://github.com/tensorflow/tensorflow/files/2093402/profile-101.json.zip)

(using feed_dict)
![lr_single_feed](https://user-images.githubusercontent.com/124190/41281927-4186bcae-6e65-11e8-80a6-e19dcd6b5c53.png)

[profile-101.json.zip](https://github.com/tensorflow/tensorflow/files/2093405/profile-101.json.zip)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 63 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by switchfootsid** [comment]
Any news on this? Facing the same issues.

**Comment by codescv** [comment]
FYI  The question is answered here:

https://stackoverflow.com/questions/50781373/using-feed-dict-is-more-than-5x-faster-than-using-dataset-api

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ymodak** [comment]
Closing this issue since 

> FYI The question is answered here:
> 
> https://stackoverflow.com/questions/50781373/using-feed-dict-is-more-than-5x-faster-than-using-dataset-api

answers the issue.
